xtras/src/supervisor.rs

```diff
@@ -17,11 +19,25 @@ pub struct Actor<T, R> {
     context: Context<T>,
     ctor: Box<dyn Fn() -> T + Send + 'static>,
     tasks: Tasks,
-    restart_policy: Box<dyn FnMut(&R) -> bool + Send + 'static>,
+    restart_policy: AsyncClosure<R>,
     _actor: Address<T>, // kept around to ensure that the supervised actor stays alive
     metrics: Metrics,
 }
 
+type AsyncClosure<R> = Box<
+    dyn for<'a> FnMut(&'a R) -> Pin<Box<dyn Future<Output = bool> + 'a + Send + Sync>>
+        + Send
+        + Sync,
+>;
+
+/// Closure that configures the supervisor to restart on every kind of error
+pub fn always_restart<E>() -> AsyncClosure<E>
+where
+    E: Error + Send + Sync + 'static,
+{
+    Box::new(|_: &E| Box::pin(async move { true }))
+}
```

Comment:
:question: Could we make this private and use `new` where necessary? I think there was a problem with that approach, but I can't figure it out when comparing the signatures of `new` and `with_policy`.

Reply:
I prefer explicit assignment of policies where the error is defined - at the end of the day, it's easy to swap one for another (end the `new()` makes a bigger diff then). I would leave this public for maximum flexibility for downstream crates.